lib/omnibus/config.rb

```diff
@@ -498,12 +498,17 @@ def reset!
 
     # The number of worker threads for make. If this is not set
     # explicitly in config, it will attempt to determine via Ohai in
-    # the builder, and failing that will default to 3
+    # the builder, and failing that will default to 3. This will
+    # also default to 3 if the count is 0 (like in a Solaris zone)
     #
     # @return [Integer]
     default(:workers) do
-      if Ohai['cpu'] && Ohai['cpu']['total']
-        Ohai['cpu']['total'].to_i + 1
+      count = 0
+      if Ohai['cpu'] && Ohai['cpu'].key?(:total)
```

Comment:
Also, though, how is 'Ohai['cpu']['total']' getting set to a value that isn't a string-of-an-integer?  If 'nil' is showing up here that's an ohai bug.